### PR TITLE
Fix Python 3.10 PR

### PR DESCRIPTION
**Pull request by simonbohnen**
- add boto3
  - provide credentials for minio via environment variables
  - set endpoint URL correctly in code
- [fix fake-gcs-server](https://github.com/simonbohnen/fake-gcs-server/commit/e0b32a7f729da0e857e973e9c92456791b8f384d) to not send checksums for partial requests (HTTP status 206) and use that version for CI
- pin setuptools-scm to 6.4.2 to be compatible with Python 3.6
  - @xhochy should we drop Python 3.6 support instead?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,6 +18,9 @@ jobs:
           POSTGRES_USER: minimalkv_test
     env:
       SIMPLEKV_CI: 1
+      AWS_ACCESS_KEY_ID: minio
+      AWS_SECRET_ACCESS_KEY: miniostorage
+      AWS_DEFAULT_REGION: us-east-1
```
Comment (janjagusch) [inline .github/workflows/ci.yml:23]: shouldn't this be automatically read from [here](https://github.com/data-engineering-collective/minimalkv/blob/main/boto_credentials.ini)? why do we need to set them explicitly?

File: docker-compose.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 services:
   gcs:
-    image: fsouza/fake-gcs-server
+    image: simonbohnen/fake-gcs-server
```
Comment (janjagusch) [inline docker-compose.yml:3]: Could you open a PR to get this merged upstream?

(Not blocking to get this PR merged for me, though).

File: environment.yml [hunk 3]
```diff
@@ -17,10 +17,11 @@ dependencies:
   - google-cloud-storage
   - azure-storage-blob
   - boto
+  - boto3
   - pymongo
   - docker-compose
   - pymysql
   - pre-commit
-  - setuptools-scm
+  - setuptools-scm==6.4.2
```
Comment (janjagusch) [inline environment.yml:25]: This pin is necessary to support Python 3.6, right? In that case, I'd suggest removing the Python 3.6 support (in a separate PR).

File: tests/bucket_manager.py [hunk 4]
```diff
@@ -45,13 +45,15 @@ def boto_bucket(
...
 
 @contextmanager
-def boto3_bucket(access_key, secret_key, host, bucket_name=None, **kwargs):
+def boto3_bucket(access_key, secret_key, host=None, bucket_name=None, **kwargs):
     import boto3
 
     name = bucket_name or "testrun-bucket-{}".format(uuid())
-    s3_client = boto3.client("s3")
+    if host == "127.0.0.1":
+        host = "http://127.0.0.1:9000"
```
Comment (janjagusch) [inline tests/bucket_manager.py:53]: Why is this necessary? Couldn't you directly pass `http://127.0.0.1:9000` to the function call?

File: tests/test_boto3_store.py [hunk 5]
```diff
@@ -21,7 +21,7 @@ def credentials(request):
     return request.param
 
 
-@pytest.yield_fixture()
+@pytest.fixture
```
Comment (janjagusch) [inline tests/test_boto3_store.py:24]: out of curiosity: what's the difference between `fixture` and `yield_fixture`, and why did you have to change it here?

**Review by janjagusch — COMMENTED** [review]
Nice work!

File: tests/test_boto3_store.py (hunk 5 above)
Comment (simonbohnen) [reply tests/test_boto3_store.py:24]: It's just [outdated](https://docs.pytest.org/en/7.1.x/yieldfixture.html), there was a warning

File: tests/bucket_manager.py (hunk 4 above)
Comment (simonbohnen) [reply tests/bucket_manager.py:53]: Was confused about the configuration file, but changing the host there seems to work 👍🏼

File: docker-compose.yml (hunk 2 above)
Comment (simonbohnen) [reply docker-compose.yml:3]: I'm rather thinking that the bug is on the side of the client SDK, see my comment [here](https://github.com/googleapis/google-resumable-media-python/issues/22#issuecomment-1263971993). I chose to modify the server instead as integrating that into our tests is easier than changing the modified client, I think.

I'm going to open a PR for the client in addition to the comment I posted.

File: environment.yml (hunk 3 above)
Comment (simonbohnen) [reply environment.yml:25]: Yep! See #49.

**Comment by simonbohnen** [comment]
The pre-commit hooks seem to fail due to the merge from a fork. They pass [here](https://github.com/simonbohnen/minimalkv/actions/runs/3182271750/jobs/5188003118).

**Comment by xhochy** [comment]
Can you remove https://github.com/data-engineering-collective/minimalkv/blob/f2012e7474d5880a8ebbf1c10d05d8885da5513f/.github/workflows/ci.yml#L56 and see whether that fixes the pre-commit?

**Comment by simonbohnen** [comment]
> Can you remove
> 
> https://github.com/data-engineering-collective/minimalkv/blob/f2012e7474d5880a8ebbf1c10d05d8885da5513f/.github/workflows/ci.yml#L56
> 
> and see whether that fixes the pre-commit?

@xhochy Thank you, that's probably the source 👌🏼

**Comment by xhochy** [comment]
Looks good except the comments that were already raised by @janjagusch.

**Comment by janjagusch** [comment]
> Looks good except the comments that were already raised by @janjagusch.

@xhochy, all my comments have been resolved, so I think this is good to get merged.

**Comment by xhochy** [comment]
> Also, should we merge straight into `main`, instead of into the `python-3.10` branch?

Yes!

**Comment by simonbohnen** [comment]
> > Also, should we merge straight into `main`, instead of into the `python-3.10` branch?
> 
> Yes!

Thx for changing that 👌🏼 Can you merge the PR?
